cscs-checks/system/basic/file_system_check.py

```diff
@@ -0,0 +1,101 @@
+# Copyright 2016-2020 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import os
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+class FileSystemCommandCheck(rfm.RunOnlyRegressionTest):
+    def __init__(self):
+        self.descr = 'File system sanity test base'
+        # TODO: test from cn as well
+        self.valid_systems = ['daint:login', 'dom:login']
+        self.valid_prog_environs = ['builtin']
+        self.num_tasks = 1
+        self.num_tasks_per_node = 1
+        self.sanity_patterns = sn.assert_found(r'0', self.stdout)
+        self.perf_patterns = {
+            'real_time': sn.extractsingle(r'\nreal.+m(?P<real_time>\S+)s',
+                                     self.stderr, 'real_time', float)
+        }
+        self.postrun_cmds = ['echo $?']
+        self.tags = {'ops', 'diagnostic'}
+        self.maintainers = ['CB']
+
+
+@rfm.parameterized_test(['/scratch/snx1*'],
+                        ['/scratch/snx3*'])
+class FileSystemChangeDirCheck(FileSystemCommandCheck):
+    def __init__(self, variant):
+        super().__init__()
+        self.descr = 'Change directory to scratch test'
+        self.reference = {
+            'daint:login': {
+                'real_time': (0.1, None, 0.1, 's')
+            },
+            'dom:login': {
+                'real_time': (0.1, None, 0.1, 's')
+            }
+        }
+        self.executable = 'time cd'
+        self.executable_opts = [variant]
+
+
+@rfm.parameterized_test(['/scratch/snx1*'],
+                        ['/scratch/snx3*'])
+class FileSystemLsDirCheck(FileSystemCommandCheck):
+    def __init__(self, variant):
+        super().__init__()
+        self.descr = 'ls of directory in scratch test'
+        self.reference = {
+            'daint:login': {
+                'real_time': (0.1, None, 0.1, 's')
+            },
+            'dom:login': {
+                'real_time': (0.1, None, 0.1, 's')
+            }
+        }
+        self.executable = 'time ls'
+        self.executable_opts = [variant]
+
+                       
+@rfm.parameterized_test(['/project/csstaff/bignamic'],#TODO: can we parametrize the user?
+                        ['/users/bignamic'],
```

Comment:
Can we use `$HOME` instead of user `/users/bignamic`? This would make the test more generic...

Reply:
I agree. Same for project, using `$PROJECT` and the user name. Use a ReFrame [utility](https://reframe-hpc.readthedocs.io/en/stable/utility_functions_reference.html#reframe.utility.osext.osuser) function to get it.

Reply:
Done